miniconf/src/tree.rs

```diff
@@ -302,9 +315,31 @@ impl Key for &str {
 /// This behavior is upheld by and compatible with all implementations in this crate. It is only violated
 /// when deriving `TreeKey` for a struct that (a) forwards its own type parameters as type
 /// parameters to its field types, (b) uses `TreeKey` on those fields, and (c) those field
-/// types use their type parameters at other levels than `TreeKey<Y - 1>`. See the
+/// types use their type parameters at other levels than `TreeKey<Y - 1>`. See also the
 /// `test_derive_macro_bound_failure` test in `tests/generics.rs`.
 ///
+/// ## Validation
+///
+/// The `TreeDeserialize` derive macro supports validation callbacks.
+/// For leaf fields the signature is `fn(struct: &S, new: T, ident: &str, old: &T) -> Result<T, &str>`
+/// The callback returns `Ok(new)` if the value is to be set. If the callback returns
+/// an `Err(&str)`, the update is aborted and the value remains unchanged.
+/// For intermediate fields the signature is `fn(new: &mut T, ident: &str) -> Result<(), &str>`.
```

Comment:
I'm not sure what this is referring to with "intermediate fields". Perhaps we can add an example below to make it clear?

Reply:
We use `non-leaf` elsewhere. Taking that.

Reply:
I still think adding an example code block to the docs may be helpful, but yeah, non-leaf would be more in-line with existing docs.